File: factory/Exemplo1/AnimalGato.java
public class AnimalGato extends Animal {
    public AnimalGato(String nome) {
		this.nome = nome;
		System.out.println("Sou o gato " + this.nome);
	} 

    public void falar() {
        System.out.println("miaaau!");
    }        
}

File: factory/Exemplo1/AnimalGato.ts
class AnimalGato extends Animal {
    constructor(nome: string) {
        super();  // Chama o construtor da classe pai (Animal)
        this.nome = nome;
        console.log("Sou o gato " + this.nome);
    }

    falar(): void {
        console.log("miaaau!");
    }
}